docs_nnx/nnx_glossary.rst

```diff
@@ -0,0 +1,43 @@
+
+    GraphDef
+      :class:`nnx.GraphDef<flax.nnx.GraphDef>` is a class that represents all the static, stateless, and Pythonic parts of a Flax :term:`Module<Module>` (:class:`nnx.Module<flax.nnx.Module>`).
+
+    Merge
+      Refer to :term:`Split and merge<Split and merge>`.
+
+    Module
+      :class:`nnx.Module <flax.nnx.Module>` is a dataclass that enables defining and initializing parameters in a referentially-transparent form. It is responsible for storing and updating :term:`Variable<Variable> objects and parameters within itself.
+
+    Params / parameters
+       :class:`nnx.Param <flax.nnx.Param>`, a particular subclass of :class:`nnx.Variable <flax.nnx.Variable>` that generally contains the trainable weights.
+
+    PRNG states
+      A Flax :class:`nnx.Module <flax.nnx.Module>` can keep a reference of a `pseudorandom number generator (PRNG) <https://jax.readthedocs.io/en/latest/random-numbers.html>`__ state object :class:`nnx.Rngs <flax.nnx.Rngs>` that can generate new JAX `PRNG <https://en.wikipedia.org/wiki/Pseudorandom_number_generator>`__ keys. These keys are used to generate random JAX arrays through `JAX's functional pseudorandom number generators <https://jax.readthedocs.io/en/latest/random-numbers.html>`__.
+      You can use a PRNG state with different seeds to add more fine-grained control to your model (for example, to have independent random numbers for parameters and dropout masks).
+      Refer to the `Flax Randomness/PRNG guide <https://flax.readthedocs.io/en/latest/guides/randomness.html>`__
+      for more details.
+
+    Split and merge
+      :meth:`nnx.split <flax.nnx.split>` is a way to represent an :class:`nnx.Module <flax.nnx.Module>` by two parts: 1) a static Flax NNX :term:`GraphDef <GraphDef>` that captures its Pythonic static information; and 2) one or more :term:`Variable state(s)<Variable state>` that capture its `JAX arrays <https://jax.readthedocs.io/en/latest/_autosummary/jax.Array.html>`__ (``jax.Array``) in the form of `JAX pytrees <https://jax.readthedocs.io/en/latest/working-with-pytrees.html>`__. They can be merged back to the original ``nnx.Module`` using :meth:`nnx.merge <flax.nnx.merge>`.
+
+    Transformation
+      A Flax NNX transformation (transform) is a Flax NNX-wrapped version of a `JAX transformation <https://flax.readthedocs.io/en/latest/guides/transforms.html>`__ that allows a transformed function to take the Flax NNX :term:`Module<Module>` (``nnx.Module``) as input or output. For example, a "lifted" version of `jax.jit <https://jax.readthedocs.io/en/latest/_autosummary/jax.jit.html#jax.jit>`__ is :meth:`nnx.jit <flax.nnx.jit>`. Check out the `Flax NNX transforms guide <https://flax.readthedocs.io/en/latest/guides/transforms.html>`__ to learn more.
```

Comment:
> Flax NNX-wrapped version

This is verbose. Just `"lifted" version` should do. I used "wrapped" previously because it's a more common word, but we use the word "lifted" more extensively too. 

> a transformed function

Use "the function being transformed" because the JAX transformation by default transforms an input function.

Reply:
Thank you, as discussed we'll use "wrapped version" instead of "Flax NNX-wrapped version.

Also adding "that is" to "being transformed".

In summary:

```
    Transformation
      A Flax NNX transformation (transform) is a wrapped version of a `JAX transformation <https://flax.readthedocs.io/en/latest/guides/transforms.html>`__ that allows the function that is being transformed to take the Flax NNX :term:`Module<Module>` (``nnx.Module``) as input or output. For example, a "lifted" version of `jax.jit <https://jax.readthedocs.io/en/latest/_autosummary/jax.jit.html#jax.jit>`__ is :meth:`nnx.jit <flax.nnx.jit>`. Check out the `Flax NNX transforms guide <https://flax.readthedocs.io/en/latest/guides/transforms.html>`__ to learn more.
```